### ISSUE TITLE
dockerd crash while in swarm mode

### ISSUE THREAD

**Issue by lbguilherme** [docker/docker#27109]
I started docker 1.12.2-rc1 in swarm mode with 5 managers and 0 workers 2 days ago, today one the servers (I believe it was the leader at the time, but I'm not sure) was down because dockerd crashed. The machines where created with docker-machine on AWS and were communicating via a private subnet there. This happened only once, I don't know how to reproduce.

I had 5 services running in a overlay network, with a total of 20 containers. Only one of those services where exposed on the ingress network (6 containers).

Here are the logs, including a backtrace:

```
Oct  3 07:17:07 server01 docker[673]: time="2016-10-03T07:17:07.298243072Z" level=warning msg="containerd: unable to save 4853747279e04d436c131a63a52f6958d4ddfc2c9cb52c7c34216dc7d0d1006e:af40500cf2648c1666bc615584b1c07240bf3c6a9de2e20ea2505c7f2614af0c starttime: open /proc/18988/stat: no such file or directory"
Oct  3 07:17:07 server01 docker[673]: time="2016-10-03T07:17:07.298840778Z" level=info msg="containerd: 4853747279e04d436c131a63a52f6958d4ddfc2c9cb52c7c34216dc7d0d1006e:af40500cf2648c1666bc615584b1c07240bf3c6a9de2e20ea2505c7f2614af0c (pid 18988) has become an orphan, killing it"
Oct  3 07:27:09 server01 docker[673]: time="2016-10-03T07:27:09.696498275Z" level=info msg="2e0d7ace5974598b [term: 24] received a MsgVote message with higher term from 27a2d87eb2aaac1f [term: 25]"
Oct  3 07:27:09 server01 docker[673]: time="2016-10-03T07:27:09.696900661Z" level=info msg="2e0d7ace5974598b became follower at term 25"
Oct  3 07:27:09 server01 docker[673]: time="2016-10-03T07:27:09.697110440Z" level=info msg="2e0d7ace5974598b [logterm: 24, index: 3424, vote: 0] voted for 27a2d87eb2aaac1f [logterm: 24, index: 3424] at term 25"
Oct  3 07:27:09 server01 docker[673]: time="2016-10-03T07:27:09.697316207Z" level=info msg="raft.node: 2e0d7ace5974598b lost leader 49917b59e0eac37c at term 25"
Oct  3 07:27:09 server01 docker[673]: time="2016-10-03T07:27:09.703271989Z" level=info msg="raft.node: 2e0d7ace5974598b elected leader 27a2d87eb2aaac1f at term 25"
Oct  3 07:27:10 server01 docker[673]: time="2016-10-03T07:27:10.181050794Z" level=error msg="agent: session failed" error="rpc error: code = 13 desc = transport is closing" module=agent
Oct  3 07:27:10 server01 docker[673]: time="2016-10-03T07:27:10.252660063Z" level=error msg="container status unavailable" error="context canceled" module=taskmanager task.id=95fb1w4r13ry77rdv64hqitkl
Oct  3 07:27:10 server01 docker[673]: time="2016-10-03T07:27:10.253014864Z" level=error msg="container status unavailable" error="context canceled" module=taskmanager task.id=e4pyu914hu9s5qbfcwcdpm0zh
Oct  3 07:27:10 server01 docker[673]: time="2016-10-03T07:27:10.253309712Z" level=error msg="container status unavailable" error="context canceled" module=taskmanager task.id=23g5uxffl1xrpbeodyfj1wadd
Oct  3 07:27:10 server01 docker[673]: time="2016-10-03T07:27:10.253561723Z" level=error msg="container status unavailable" error="context canceled" module=taskmanager task.id=3qkkv8grze2adaff2yng9lkgy
Oct  3 07:27:10 server01 docker[673]: time="2016-10-03T07:27:10.290492886Z" level=info msg="2016/10/03 07:27:10 [INFO] memberlist: Suspect server05-ac695dfdd498 has failed, no acks received\n"
Oct  3 07:27:12 server01 docker[673]: time="2016-10-03T07:27:12.207067328Z" level=info msg="2016/10/03 07:27:12 [INFO] memberlist: Marking server05-ac695dfdd498 as failed, suspect timeout reached\n"
Oct  3 07:27:16 server01 docker[673]: time="2016-10-03T07:27:16.053591916Z" level=info msg="2e0d7ace5974598b [term: 25] ignored a MsgHeartbeat message with lower term from 49917b59e0eac37c [term: 24]"
Oct  3 07:27:16 server01 docker[673]: time="2016-10-03T07:27:16.096262653Z" level=warning msg="2016/10/03 07:27:16 [WARN] memberlist: Refuting a suspect message (from: server05-ac695dfdd498)\n"
Oct  3 07:27:16 server01 docker[673]: time="2016-10-03T07:27:16.098151592Z" level=info msg="2e0d7ace5974598b [term: 25] ignored a MsgHeartbeat message with lower term from 49917b59e0eac37c [term: 24]"
Oct  3 07:27:16 server01 docker[673]: time="2016-10-03T07:27:16Z" level=info msg="Firewalld running: false"
Oct  3 07:27:16 server01 docker[673]: message repeated 3 times: [ time="2016-10-03T07:27:16Z" level=info msg="Firewalld running: false"]
Oct  3 07:27:16 server01 docker[673]: panic: close of closed channel
Oct  3 07:27:16 server01 docker[673]: goroutine 939999 [running]:
Oct  3 07:27:16 server01 docker[673]: panic(0x1700a60, 0xc823291b90)
Oct  3 07:27:16 server01 docker[673]: #011/usr/local/go/src/runtime/panic.go:481 +0x3e6
Oct  3 07:27:16 server01 docker[673]: github.com/docker/libnetwork/networkdb.(*NetworkDB).handleBulkSync(0xc820793b00, 0xc821d28000, 0x3738, 0x4000)
Oct  3 07:27:16 server01 docker[673]: #011/usr/src/docker/vendor/src/github.com/docker/libnetwork/networkdb/delegate.go:325 +0x2c1
Oct  3 07:27:16 server01 docker[673]: github.com/docker/libnetwork/networkdb.(*NetworkDB).handleMessage(0xc820793b00, 0xc822298000, 0x373d, 0x373d, 0x3700)
Oct  3 07:27:16 server01 docker[673]: #011/usr/src/docker/vendor/src/github.com/docker/libnetwork/networkdb/delegate.go:358 +0x377
Oct  3 07:27:16 server01 docker[673]: github.com/docker/libnetwork/networkdb.(*delegate).NotifyMsg(0xc820844150, 0xc822298000, 0x373d, 0x373d)
Oct  3 07:27:16 server01 docker[673]: #011/usr/src/docker/vendor/src/github.com/docker/libnetwork/networkdb/delegate.go:371 +0x52
Oct  3 07:27:16 server01 docker[673]: github.com/hashicorp/memberlist.(*Memberlist).readUserMsg(0xc820afefc0, 0x7fdb203204a0, 0xc82199fa10, 0xc821777740, 0x0, 0x0)
Oct  3 07:27:16 server01 docker[673]: #011/usr/src/docker/vendor/src/github.com/hashicorp/memberlist/net.go:987 +0x389
Oct  3 07:27:16 server01 docker[673]: github.com/hashicorp/memberlist.(*Memberlist).handleConn(0xc820afefc0, 0xc822c361b0)
Oct  3 07:27:16 server01 docker[673]: #011/usr/src/docker/vendor/src/github.com/hashicorp/memberlist/net.go:216 +0x1912
Oct  3 07:27:16 server01 docker[673]: created by github.com/hashicorp/memberlist.(*Memberlist).tcpListen
Oct  3 07:27:16 server01 docker[673]: #011/usr/src/docker/vendor/src/github.com/hashicorp/memberlist/net.go:196 +0x163
```

docker version:
```
Client:
 Version:      1.12.2-rc1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   45bed2c
 Built:        Tue Sep 27 21:29:42 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.2-rc1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   45bed2c
 Built:        Tue Sep 27 21:29:42 2016
 OS/Arch:      linux/amd64
```

docker info: (after restarting the daemon)
```
Containers: 20
 Running: 0
 Paused: 0
 Stopped: 20
Images: 26
Server Version: 1.12.2-rc1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 101
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay host bridge null
Swarm: active
 NodeID: boq2xo0lo5o10zdllt5c8z0zs
 Is Manager: true
 ClusterID: eir1c28uzqvpjfn1rqeu9n774
 Managers: 5
 Nodes: 5
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.7.94.21
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-38-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 990.7 MiB
Name: server01
ID: USJC:HW4V:NKK6:MWHX:N7YO:OWJK:SJ6B:U2JQ:7DPZ:PGY2:IVME:K5W7
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Labels:
 provider=amazonec2
Insecure Registries:
 127.0.0.0/8
```

**Comment by cpuguy83** [comment]
ping @mrjana 
@vieux

**Comment by cpuguy83** [comment]
offending code is here:

```go
	if !bsm.Unsolicited {
		nDB.RLock()
		ch, ok := nDB.bulkSyncAckTbl[bsm.NodeName]
		nDB.RUnlock()
		if ok {
			close(ch)
		}

		return
	}
```

**Comment by mrjana** [comment]
@lbguilherme Thanks for the report. Will fix it.

**Comment by lbguilherme** [comment]
Fixed by #27120
